### DISCUSSION TITLE
Alter json body in POST request

### DISCUSSION THREAD

**Discussion by eNTi** [canton7/RestEase#205] [Q&A] [answered]
Hi,
I've a particular issue. I need the request I'm sending enclosed in the following:
```
{
    req:
    {
        ... default body ...
    }
}
```
Is this somehow possible?

**Comment by canton7** [comment, marked as answer]
You have two options:

1. Create an e.g. `Request<T>` type which wraps each body you send, e.g:

```cs
public class Request<T>
{
    [JsonProperty("req")]
    public T Body { get; set; }

    public Request(T body) => Body = body;
}

[Post(...)]
public void PostAsync([Body] Request<SomeBody> body);
```

2. Write your own [RequestBodySerializer](https://github.com/canton7/RestEase#serializing-request-bodies-requestbodyserializer) which wraps the serialized body in the containing object. See that link for more details -- it will look like [this](https://github.com/canton7/RestEase/blob/master/src/RestEase/JsonRequestBodySerializer.cs), but wrapped in an extra JSON object.

**Reply by eNTi** [reply]
Thank you.

**Reply by eNTi** [reply]
Could you elaborate where I would use PostAsync? Where do I put that so that it is picked up by the RestClient.For?

**Reply by canton7** [reply]
That's an example of the method to put in your interface (the one you pass to `RestClient.For<T>`). Just to demonstrate that the body type is `Request<Something>`

**Reply by eNTi** [reply]
Ah. I think I've got it. Thanks again.
